backend/src/v2/compiler/visitor.go

```diff
@@ -111,11 +111,13 @@ func (state *pipelineDFS) dfs(name string, component *pipelinespec.ComponentSpec
 
 		// Add kubernetes spec to annotation
 		if state.kubernetesSpec != nil && state.kubernetesSpec.DeploymentSpec != nil {
-			kubernetesExecSpec, ok := state.kubernetesSpec.DeploymentSpec.Executors[executorLabel]
-			if ok {
-				err := state.visitor.AddKubernetesSpec(name, kubernetesExecSpec)
-				if err != nil {
-					return componentError(fmt.Errorf("failed to add Kubernetes spec for %s: %w", name, err))
+			if state.kubernetesSpec.DeploymentSpec.Executors != nil {
```

Comment:
I don't think this is needed since nil maps in Go can still be accessed safely but it's okay to have it here too.

Reply:
I added the nil check as I encountered a panic during testing. I believe the root cause is that Executors is a protobuf-generated map (map[string]*structpb.Struct).